hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java

```diff
@@ -198,7 +198,7 @@ protected NettyChannelBuilder createChannel(DatanodeDetails dn, int port)
         NettyChannelBuilder.forAddress(dn.getIpAddress(), port).usePlaintext()
             .maxInboundMessageSize(OzoneConsts.OZONE_SCM_CHUNK_MAX_SIZE)
             .intercept(new GrpcClientInterceptor());
-    if (secConfig.isGrpcTlsEnabled()) {
+    if (secConfig.isSecurityEnabled() && secConfig.isGrpcTlsEnabled()) {
```

Comment:
Do we know why we have two config for this thing? If it is historical then I would not change the semantics here, but in a separate JIRA so that we might have extra opinions on this one.

I remember that it caused problems when just security was enabled and the grpc tls flag was turned off, but I am curious why we did not fixed it here. @kerneltime @swagle do you have any memories on this?

Reply:
@fapifta but enabling TLS cannot be assumed with security a.k.a kerberos in the Hadoop world. TLS has definite perf overhead hence these settings we separate similar to rpc.encryption setting in HDFS. Many production setups have the first but not the other.

Reply:
For the Om Grpc server and s3g client, we use enable TLS (`secConf.isGrpcTlsEnabled()`) only with security enabled, otherwise it logs the error a creates a Grpc "plaintext" channel.  Similar checking for TLS enabled and security as is here.
https://github.com/apache/ozone/blob/5db6fc50ba851d90871173b49d96991fe6f068dc/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/GrpcOzoneManagerServer.java#L101

Reply:
> For the Om Grpc server and s3g client, we use enable TLS (`secConf.isGrpcTlsEnabled()`) only with security enabled, otherwise it logs the error a creates a Grpc "plaintext" channel. Similar checking for TLS enabled and security as is here.
> 
> https://github.com/apache/ozone/blob/5db6fc50ba851d90871173b49d96991fe6f068dc/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/GrpcOzoneManagerServer.java#L101

Thanks @neils-dev  for pointing out this space.  It is going to be addressed in my next patch [HDDS-7590](https://issues.apache.org/jira/browse/HDDS-7590)

Reply:
Thanks for the inputs, I have initiated a broader discussion so that we can see clear what is with these configs, but in the meantime I support to commit this patch as it is.